### PR TITLE
refactor(client/job): more polishing

### PR DESCRIPTION
**Pull request by Manason**
**Describe Pull request**
- added function headers
- extracted common function for taking out vehicles
- extracted common function for elevators
- made GetClosestPlayer global and moved to main for use in multiple files.
- light general refactoring/cleanup.

PR to be reviewed after #23 as it's built on top.

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? [no] (Be honest)
- Does your code fit the style guidelines? [yes]
- Does your PR fit the contribution guidelines? [yes]

### CODE REVIEW

File: client/job.lua [hunk 1]
```diff
@@ -1,39 +1,56 @@
...
     QBCore.Functions.TriggerCallback('QBCore:Server:SpawnVehicle', function(netId)
         local veh = NetToVeh(netId)
-        SetVehicleNumberPlateText(veh, Lang:t('info.amb_plate') .. tostring(math.random(1000, 9999)))
-        SetEntityHeading(veh, coords.w)
-        SetVehicleFuelLevel(veh, 100.0)
-        TaskWarpPedIntoVehicle(cache.ped, veh, -1)
-        if Config.VehicleSettings[vehicleInfo] ~= nil then
-            QBCore.Shared.SetDefaultVehicleExtras(veh, Config.VehicleSettings[vehicleInfo].extras)
+        takeOutVehicle(veh, Lang:t('info.amb_plate'), coords.w)
+        if Config.VehicleSettings[vehicleName] ~= nil then
```
Comment (YvanCywan) [inline client/job.lua]:
```suggestion
        if Config.VehicleSettings[vehicleName] then
```

File: client/job.lua [hunk 2]
```diff
@@ -45,75 +62,81 @@ local function menuGarage()
...
-    SetPedArmour(ped, playerData.metadata["armor"])
+    SetPedArmour(ped, playerMetadata.armor)
 end
 
-local function initDeathAndLastStand(playerData)
-    if (not playerData.metadata["inlaststand"] and playerData.metadata["isdead"]) then
+---starts death or last stand based off of player's metadata
+---@param metadata any
+local function initDeathAndLastStand(metadata)
+    if (not metadata.inlaststand and metadata.isdead) then
```
Comment (YvanCywan) [inline client/job.lua]:
```suggestion
    if not metadata.inlaststand and metadata.isdead then
```

File: client/job.lua [hunk 3]
```diff
@@ -45,75 +62,81 @@ local function menuGarage()
...
-        initHealthAndArmor(ped, playerId, playerData)
-        initDeathAndLastStand(playerData)
-        if PlayerJob.name == 'ambulance' and PlayerJob.onduty then
-            TriggerServerEvent("hospital:server:AddDoctor", PlayerJob.name)
+        playerJob = playerData.job
+        initHealthAndArmor(ped, playerId, playerData.metadata)
+        initDeathAndLastStand(playerData.metadata)
+        if playerJob.name == 'ambulance' and playerJob.onduty then
+            TriggerServerEvent("hospital:server:AddDoctor", playerJob.name)
         end
```
Comment (YvanCywan) [inline client/job.lua]:
```suggestion
        if playerJob.name ~= 'ambulance' or not playerJob.onduty then return end
        TriggerServerEvent("hospital:server:AddDoctor", playerJob.name)
```

File: client/job.lua [hunk 4]
```diff
@@ -45,75 +62,81 @@ local function menuGarage()
...
     end)
 end)
 
+---Update doctor count.
 RegisterNetEvent('QBCore:Client:OnPlayerUnload', function()
-    if PlayerJob.name == 'ambulance' and PlayerJob.onduty then
-        TriggerServerEvent("hospital:server:RemoveDoctor", PlayerJob.name)
+    if playerJob.name == 'ambulance' and playerJob.onduty then
+        TriggerServerEvent("hospital:server:RemoveDoctor", playerJob.name)
     end
```
Comment (YvanCywan) [inline client/job.lua]:
```suggestion
    if playerJob.name ~= 'ambulance' or not playerJob.onduty then return end
    TriggerServerEvent("hospital:server:RemoveDoctor", playerJob.name)
```

File: client/damage/damage.lua [hunk 5]
```diff
@@ -129,9 +129,8 @@ local function applyDamage(ped, damageDone, isArmorDamaged)
             end
         end
     elseif Config.AlwaysBleedChanceWeapons[weapon]
-        and math.random(100) < Config.AlwaysBleedChance
+        and math.random(100) < Config.AlwaysBleedChance
```
Comment (BerkieBb) [inline client/damage/damage.lua]: WHITESPACE?????#!?!??!?

File: client/job.lua (hunk 2 above)
Comment (BerkieBb) [inline client/job.lua]: () useless

File: client/job.lua [hunk 6]
```diff
@@ -45,75 +62,81 @@ local function menuGarage()
...
-    if (not playerData.metadata["inlaststand"] and playerData.metadata["isdead"]) then
+---starts death or last stand based off of player's metadata
+---@param metadata any
+local function initDeathAndLastStand(metadata)
+    if (not metadata.inlaststand and metadata.isdead) then
         deathTime = Laststand.ReviveInterval
         OnDeath()
         DeathTimer()
-    elseif (playerData.metadata["inlaststand"] and not playerData.metadata["isdead"]) then
+    elseif (metadata.inlaststand and not metadata.isdead) then
```
Comment (BerkieBb) [inline client/job.lua]: () useless

File: client/job.lua [hunk 7]
```diff
@@ -131,8 +154,12 @@ local function status()
     lib.showContext('ambulance_status_context_menu')
 end
 
---- TODO: Refactor
-local function getPlayerStatus(k, v, result)
+---notify doctor in chat of a patient's health status.
+---TODO: Refactor
+---@param k any
```
Comment (BerkieBb) [inline client/job.lua]: not any, string

File: client/job.lua [hunk 8]
```diff
@@ -131,8 +154,12 @@ local function status()
...
+---@param v any
```
Comment (BerkieBb) [inline client/job.lua]: v would be table here

File: client/job.lua [hunk 9]
```diff
@@ -131,8 +154,12 @@ local function status()
...
+---@param result any
```
Comment (BerkieBb) [inline client/job.lua]: result would also be table

**Review by BerkieBb — CHANGES_REQUESTED** [review]

File: client/job.lua [hunk 10]
```diff
@@ -250,159 +280,127 @@ RegisterNetEvent('hospital:client:TreatWounds', function()
...
     end)
 end
 
+---Opens the hospital stash.
 RegisterNetEvent('qb-ambulancejob:stash', function()
-    if PlayerJob.onduty then
+    if playerJob.onduty then
         TriggerServerEvent("inventory:server:OpenInventory", "stash", "ambulancestash_" .. QBCore.Functions.GetPlayerData().citizenid)
         TriggerEvent("inventory:client:SetCurrentStash", "ambulancestash_" .. QBCore.Functions.GetPlayerData().citizenid)
     end
```
Comment (YvanCywan) [inline client/job.lua]:
```suggestion
    if not playerJob.onduty then return end
    TriggerServerEvent("inventory:server:OpenInventory", "stash", "ambulancestash_" .. QBCore.Functions.GetPlayerData().citizenid)
    TriggerEvent("inventory:client:SetCurrentStash", "ambulancestash_" .. QBCore.Functions.GetPlayerData().citizenid)
```

**Review by YvanCywan — CHANGES_REQUESTED** [review]

File: client/job.lua (hunk 8 above)
Comment (Manason) [reply client/job.lua]: or an integer in the case of result["BLEED"]. Updated.

**Review by YvanCywan — APPROVED** [review]

**Review by BerkieBb — APPROVED** [review]

**Pull request merged**